pages/api/v1/stats.ts

```diff
@@ -0,0 +1,35 @@
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+import { connectToDatabase } from '../../../server/mongoDB';
+import { protect } from '../../../server/protect';
+
+const statsHandler: NextApiHandler = async (req, res) => {
+  switch (req.method) {
+    case 'GET':
+      await getStats(req, res);
+      break;
+    default:
+      return res.status(405).setHeader('Allow', 'GET').send(undefined);
+  }
+};
+
+const getStats: NextApiHandler = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  const { userDataCollection } = await connectToDatabase();
```

Comment:
add a check isAdmin() before connecting to database. maybe worth creating another higher-order handler (similar to `protect`)